### PR TITLE
Translate random messages

### PR DESCRIPTION
**Pull request by Lucyucy**
Adds german translations for random messages. More translations are welcome! :)

### CODE REVIEW

**Comment by magnetar47** [comment]
Please don't add multiple things in one PR. If this PR adds translation support, then it should be focused on that only. Multiple things in one PR is generally not good in terms of reviewing, etc. Let's say the translation part is approved but the new rmessage isn't. Then the approved one alone can't be merged, as they're all one PR. Do you see the issue? Separate PRs for separate things is always better. :)

**Comment by magnetar47** [comment]
Fixed the merge conflict caused by #1042

File: mods/other/random_messages/locale/random_messages.de.tr [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+For example: /top50 mode:nade_fight.=Zum Beispiel: /top50 mode:nade_fight.
+To check someone's rank on another mode use /r <mode:technical modename> <playername>.=Um den Platz eines anderen Spielers in einem anderen Mode nachzuschauen nutze /r <mode:technical modename> <playername>.
+For example: /r mode:nade_fight randomplayer."=Zum Beispiel: /r mode:nade_fight randomplayer
+To check someone's team use /team player <player_name>.=Um das Team eines Spielers nachzuschauen, nutze /team player <player_name>
+To check all team members use /team.=Um alle Teammitglieder nachzuschauen, nutze /team.
+You can capture multiple enemy flags at once!=Du kannst mehrere Flaggen zugleich holen!
+Consider joining our Discord server at https://discord.gg/vcZTRPX=Tritt unserem Discord Server bei unter https://discord.gg/vcZTRPX
+You can press sneak while jumping, to jump up two blocks.=Halte Sneak gedrückt während du springst, um zwei Blöcke hochzuspringen.
+Use /donate <playername> <score> to reward a team-mate for their work.=Nutze /donate <playername> <amount>, um ein Teammitglied für seine Arbeit zu belohnen.
```
Comment (magnetar47) [inline mods/other/random_messages/locale/random_messages.de.tr:36]:
```suggestion
Use /donate <playername> <score> to reward a team-mate for their work.=Nutze /donate <playername> <score>, um ein Teammitglied für seine Arbeit zu belohnen.
```
You probably meant to change it here as well? :)

**Comment by Panquesito7** [comment]
Once the PR is merged (hopefully), I'll work on the Spanish translation. 🙂

**Comment by magnetar47** [comment]
@Lucyucy Is this PR complete? Are you planning to add anything else to this?

**Comment by Lucyucy** [comment]
I'm working on an italian translation, but if it's better to merge it before we add more translations then it is complete

**Comment by magnetar47** [comment]
> I'm working on an italian translation, but if it's better to merge it before we add more translations then it is complete

Agreed. That should be a separate PR.

**Review by LoneWolfHT — COMMENTED** [review]
Would this still work if you deleted all the `S()` in `random_messages.read_messages()` and used a single one each for `random_messages.display_message()` and related functions?

**Comment by magnetar47** [comment]
> Would this still work if you deleted all the `S()` in `random_messages.read_messages()` and used a single one each for `random_messages.display_message()` and related functions?

Oh, can it work like that? That's good to know.

**Comment by olliy1or** [comment]
> > Would this still work if you deleted all the `S()` in `random_messages.read_messages()` and used a single one each for `random_messages.display_message()` and related functions?
> 
> Oh, can it work like that? That's good to know.

can by translating the message when its sent
```lua
minetest.chat_send_all(S(msg))

**Comment by magnetar47** [comment]
> can by translating the message when its sent
> 
> ```lua
> minetest.chat_send_all(S(msg))
> ```

Neat. Thought only static strings can be wrapped by `S(...)`. So I'm guessing it can only translate messages that have translations available. What would happen if a string that's passed to `S()` doesn't have any translation available?

**Comment by Thomas--S** [comment]
> Thought only static strings can be wrapped by S(...).

Afaik, it works just fine for any strings (including variables).

The advantage of using only static strings is that it is easier to write tools for extracting translatable strings.

**Comment by Lucyucy** [comment]
> > > Would this still work if you deleted all the `S()` in `random_messages.read_messages()` and used a single one each for `random_messages.display_message()` and related functions?
> > 
> > 
> > Oh, can it work like that? That's good to know.
> 
> can by translating the message when its sent
> 
> ```lua
> minetest.chat_send_all(S(msg))
> ```

It does not work for me

File: mods/other/random_messages/init.lua [hunk 2]
```diff
@@ -70,7 +72,7 @@ function random_messages.read_messages()
 end
 
 function random_messages.display_message(message_number)
-	local msg = random_messages.messages[message_number] or message_number
+	local msg = S(random_messages.messages[message_number] or message_number)
 	if msg then
 		minetest.chat_send_all(minetest.colorize("#808080", msg))
```
Comment (LoneWolfHT) [inline mods/other/random_messages/init.lua:77]:
```suggestion
	local msg = random_messages.messages[message_number] or message_number
	if msg then
		minetest.chat_send_all(minetest.colorize("#808080", S(msg)))
```

**Review by LoneWolfHT — COMMENTED** [review]
Looks good

File: mods/other/random_messages/init.lua (hunk 2 above)
Comment (Panquesito7) [reply mods/other/random_messages/init.lua:77]: Note: if the sent message contains a newline (or `/n`), it won't be colorized for some reason. It has happened to me when translating using this way. Not sure what's the cause, though.

File: mods/other/random_messages/init.lua [hunk 3]
```diff
@@ -57,7 +58,8 @@ function random_messages.read_messages()
 		"Strengthen your team by capturing enemy flags.",
 		"Hitting your enemy does more damage than not hitting them.",
 		"Use /top50 command to see the leaderboard.",
-		"Use /top50 <mode:technical modename> to see the leaderboard on another mode. For example: /top50 mode:nade_fight.",
+		"Use /top50 <mode:technical modename> to see the leaderboard on another mode."
+		.. "For example: /top50 mode:nade_fight.",
```
Comment (LoneWolfHT) [inline mods/other/random_messages/init.lua:62]:
```suggestion
		.. " For example: /top50 mode:nade_fight.",
```

**Pull request merged**